src/basic/unit-name.c

```diff
@@ -644,13 +649,15 @@ int unit_name_mangle_with_suffix(const char *name, UnitNameMangle allow_globs, c
         if (!s)
                 return -ENOMEM;
 
-        t = do_escape_mangle(name, allow_globs, s);
-        *t = 0;
+        mangled = do_escape_mangle(name, options & UNIT_NAME_MANGLE_GLOB, s);
+        if (mangled && (options & UNIT_NAME_MANGLE_WARN))
+                log_notice("Invalid unit name \"%s\" was escaped as \"%s\" (maybe you should use systemd-escape?)",
```

Comment:
nowadays I think flags like this should just swicth between LOG_DEBUG and higher log levels, but never turn off logging entirely. i.e. I think this would be better as:

```c
log_full(options & UNIT_NAME_MANGLE_WARN ? LOG_NOTICE : LOG_DEBUG, …)
```